pkg/controller/drivelvgcontroller/drivelvgcontroller.go

```diff
@@ -0,0 +1,365 @@
+		return d.handleDriveIsNotGood(ctx, drive.Spec)
+	case !isClean:
+		return d.createACIfNotExistOrUpdate(ctx, drive.Spec)
+	}
+	return ctrl.Result{RequeueAfter: time.Minute}, nil
+}
+
+// createACIfNotExistOrUpdate tries to create AC for drive or update its size if AC already exists
+func (d *Controller) createACIfNotExistOrUpdate(ctx context.Context, drive api.Drive) (ctrl.Result, error) {
+	log := d.log.WithFields(logrus.Fields{
+		"method": "createACIfNotExistOrUpdate",
+	})
+	driveUUID := drive.GetUUID()
+	size := drive.GetSize()
+	// if drive is not clean, size is 0
+	if !drive.GetIsClean() {
+		size = 0
+	}
+	ac, err := d.cachedCrHelper.GetACByLocation(driveUUID)
+	switch {
+	case err == nil:
+		lvg, err := d.cachedCrHelper.GetLVGByDrive(ctx, driveUUID)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		// If LVG exists for Drive, we delete drive AC, because CSI uses LVG AC
+		if lvg != nil {
+			if err := d.client.DeleteCR(context.WithValue(ctx, base.RequestUUID, ac.Name), ac); err != nil {
+				log.Errorf("Error during update AvailableCapacity request to k8s: %v, error: %v", ac, err)
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, err
+		}
+		// If ac is exists, update its size to drive size
+		if ac.Spec.Size != size {
+			ac.Spec.Size = size
+			if err := d.client.Update(context.WithValue(ctx, base.RequestUUID, ac.Name), ac); err != nil {
+				log.Errorf("Error during update AvailableCapacity request to k8s: %v, error: %v", ac, err)
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{RequeueAfter: RequeueDriveTime}, nil
+	case err == errTypes.ErrorNotFound:
+		name := uuid.New().String()
+		if lvg, err := d.cachedCrHelper.GetLVGByDrive(ctx, driveUUID); err != nil || lvg != nil {
+			return ctrl.Result{}, err
+		}
+		capacity := &api.AvailableCapacity{
+			Size:         size,
+			Location:     driveUUID,
+			StorageClass: util.ConvertDriveTypeToStorageClass(drive.GetType()),
+			NodeId:       drive.GetNodeId(),
+		}
+		newAC := d.client.ConstructACCR(name, *capacity)
+		if err := d.client.CreateCR(context.WithValue(ctx, base.RequestUUID, name), name, newAC); err != nil {
+			log.Errorf("Error during create AvailableCapacity request to k8s: %v, error: %v",
+				capacity, err)
+			return ctrl.Result{}, err
+		}
+	default:
+		log.Infof("Failed to read AvailableCapacity for drive %s: %v", driveUUID, err)
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{RequeueAfter: RequeueDriveTime}, nil
+}
+
+// handleDriveIsNotGood deletes AC for bad Drive
+func (d *Controller) handleDriveIsNotGood(ctx context.Context, drive api.Drive) (ctrl.Result, error) {
+	log := d.log.WithFields(logrus.Fields{
+		"method": "handleDriveIsNotGood",
+	})
+	ac, err := d.cachedCrHelper.GetACByLocation(drive.GetUUID())
+	switch {
+	case err == nil:
+		log.Infof("Removing AC %s based on unhealthy location %s", ac.Name, ac.Spec.Location)
+		if err := d.client.DeleteCR(ctx, ac); err != nil {
```

Comment:
You need to set size to 0 - not remove

Reply:
It's the current implementation for unhealthy drives. I'll fix it then